bittensor/subtensor.py

```diff
@@ -1858,26 +1943,23 @@ def neuron_for_uid_lite(
         """
         if uid == None:
             return NeuronInfoLite._null_neuron()
+        
+        hex_bytes_result = self.query_runtime_api(
+            runtime_api="NeuronInfoRuntimeApi",
+            method="get_neuron_lite",
+            params={ 
+                "netuid": netuid,
+                "uid": uid,
+            },
+            block=block,
+        )
 
-        @retry(delay=2, tries=3, backoff=2, max_delay=4)
-        def make_substrate_call_with_retry():
-            with self.substrate as substrate:
-                block_hash = None if block == None else substrate.get_block_hash(block)
-                params = [netuid, uid]
-                if block_hash:
-                    params = params + [block_hash]
-                return substrate.rpc_request(
-                    method="neuronInfo_getNeuronLite",  # custom rpc method
-                    params=params,
-                )
-
-        json_body = make_substrate_call_with_retry()
-        result = json_body["result"]
-
-        if result in (None, []):
+        if hex_bytes_result == None:
             return NeuronInfoLite._null_neuron()
+        
+        bytes_result = bytes.fromhex(hex_bytes_result[2:])
```

Comment:
Is this just removing the byte header? I'm always weary of hard-coded list indexing, but if it's a constant than I'm fine with it.  Would it be helpful to have a try block here?